apps/astarte_pairing/lib/astarte_pairing/queries.ex

```diff
@@ -21,172 +21,190 @@ defmodule Astarte.Pairing.Queries do
   This module is responsible for the interaction with the database.
   """
 
-  alias CQEx.Query
-  alias CQEx.Result
-
   require Logger
 
+  alias Astarte.Core.Realm
+
+  @typedoc """
+  A string representing the current context, to include in error messages.
+  Not included if nil.
+  Default: nil
+  """
+  @type custom_context() :: String.t() | nil
+
+  @typedoc """
+  The type to cast the result in.
```

Comment:
The fact that the type is casted is an implementation detail that could be hidden